metron-platform/metron-storm-kafka-override/src/main/java/org/apache/storm/kafka/spout/internal/Timer.java

```diff
@@ -0,0 +1,58 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.storm.kafka.spout.internal;
+
+import java.util.concurrent.TimeUnit;
+import org.apache.storm.utils.Time;
+
+public class Timer {
+  private final long delay;
+  private final long period;
+  private final TimeUnit timeUnit;
+  private final long periodNanos;
+  private long start;
+
+  public Timer(long delay, long period, TimeUnit timeUnit) {
+    this.delay = delay;
+    this.period = period;
+    this.timeUnit = timeUnit;
+    this.periodNanos = timeUnit.toNanos(period);
+    this.start = System.nanoTime() + timeUnit.toNanos(delay);
+  }
+
+  public long period() {
+    return this.period;
+  }
+
+  public long delay() {
+    return this.delay;
+  }
+
+  public TimeUnit getTimeUnit() {
+    return this.timeUnit;
+  }
+
+  public boolean isExpiredResetOnTrue() {
```

Comment:
It is precisely the 1.1.0 version of the code with the exception of calling `System.nanoTime` instead of `Time.nanoTime` (see [here](https://github.com/apache/storm/blob/a4afacd9617d620f50cf026fc599821f7ac25c79/external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/internal/Timer.java) ).

It should be noted that `Time.nanoTime` calls `System.nanoTime` (see [here](https://github.com/apache/storm/blob/a4afacd9617d620f50cf026fc599821f7ac25c79/storm-client/src/jvm/org/apache/storm/utils/Time.java#L162) ) in a production scenario (i.e. non-simulation).